gpii/node_modules/gpii-oauth2/gpii-oauth2-authz-server/src/AuthServer.js

```diff
@@ -478,12 +550,19 @@ gpii.oauth2.authServer.contributeRouteHandlers = function (that, oauth2orizeServ
         login.ensureLoggedIn("/login"),
         function (req, res) {
             if (req.is("application/json")) {
-                var userId = req.user.id;
+                // TODO: Update the user interface to support multiple
+                // tokens per user rather than using a single default
+                var gpiiToken = req.user.defaultGpiiToken;
+
                 var oauth2ClientId = req.body.oauth2ClientId;
                 var selectedPreferences = req.body.selectedPreferences;
                 // TODO validate selectedPreferences?
-                that.authorizationService.addAuthorization(userId, oauth2ClientId, selectedPreferences);
-                res.sendStatus(200);
+                var addPromise = that.authorizationService.addAuthorization(gpiiToken, oauth2ClientId, selectedPreferences);
+                addPromise.then(function () {
```

Comment:
Refactor all these to use a common utility mapping a promise to a response handler. Or else refactor to use Kettle :)